plugins/kotlinx-serialization/testData/boxIr/delegatedProperty.kt

```diff
@@ -0,0 +1,86 @@
+// TARGET_BACKEND: JVM_IR
+
+// WITH_STDLIB
+
+import kotlinx.serialization.*
+import kotlinx.serialization.descriptors.*
+import kotlinx.serialization.encoding.*
+import kotlinx.serialization.json.*
+import kotlin.reflect.KProperty
+import kotlin.properties.*
+
+@Serializable
+data class SimpleDTO(
+    val realProp: Int,
+) {
+    @Transient
+    private val additionalProperties: Map<String, Int> = mapOf("delegatedProp" to 123)
+    @Transient
```

Comment:
I think `@Transient` shouldn't be here, as delegated property is transient automatically?